sale_coupon_advanced/wizard/sale_coupon_generate.py

```diff
@@ -0,0 +1,18 @@
+from odoo import fields, models
+
+
+class SaleCouponGenerate(models.TransientModel):
+    """Extend to add custom coupon code generation options."""
+
+    _inherit = "sale.coupon.generate"
+
+    custom_code_generation = fields.Boolean("Custom Code Generation")
+    custom_code_prefix = fields.Char("Custom Code Prefix")
+    custom_code_nbr = fields.Integer("Number of Alphanumeric symbols", default=5)
+
+    def generate_coupon(self):
```

Comment:
If you add two new fields in a program, as this wizard is meant to be called in a program context, you could directly get these needed values for your purpose. And not use a context anymore (plus it's much much readable using the datamodel than using the context imo)